libraries/AP_HAL_ChibiOS/hwdef/CarbonixCommon/scripts/cx_modules/esc.lua

```diff
@@ -0,0 +1,171 @@
+local cx_msg = require("scripts.cx_utils.cx_msg")
+local cx_aircraft = require("scripts.cx_utils.cx_aircraft")
+
+local ESC = {}
+ESC.__index = ESC
+
+-- Constructor
+function ESC.new()
+    local self = setmetatable({}, ESC)
```

Comment:
I think we should replace these with this simpler pattern. Should have no downsides as long as we never plan to have inheritance or non-singleton modules.
```
local cx_esc = {
    name = "ESC",
    number_of_esc = 5,
    ESC_WARMUP_TIME = 3000,
    ESC_RPM_THRESHOLD = 10,
    SERVO_OUT_THRESHOLD = 1010,
}

function cx_esc:init()
    print("Initializing ESC module")
end

function cx_esc:update()
    print("Updating ESC module")
end

return cx_esc
```

This will allow us to scrap the calling of `my_new_bit_module.new()` and just have `my_new_bit_module = requires(...)`

Reply:
done